harmonic/install_osx_src.md

```diff
@@ -0,0 +1,251 @@
+# Source Installation on macOS
+
+This tutorial will work for macOS Catalina 10.15 and macOS BigSur 10.16.
+
+## Install tools
+
+The use of some additional tools is recommended to help with the source compilation,
+although other ways of correctly getting and building the sources are also possible.
+
+The easiest way to get the sources of all libraries is to use
+[vcstool](https://github.com/dirk-thomas/vcstool).
+
+To compile all the different libraries and gz-sim in the right order
+[colcon](https://colcon.readthedocs.io/en/released/) is recommended.
+The colcon tool is available on all platforms using pip (or pip3, if pip fails).
+
+## Python3 from homebrew
+
+Tools and dependencies for Harmonic can be installed using the [homebrew package manager](https://brew.sh/).
+The homebrew tool can be installed by entering the following in a terminal:
+
+```bash
+/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
+```
+
+Harmonic is compatible with Python3; it can be installed by running the following in a terminal:
+
+```bash
+brew install python3
+```
+
+## vcstool and colcon from pip
+
+PIP is available on all platforms:
+
+```bash
+python3 -m pip install vcstool
+```
+
+```bash
+python3 -m pip install -U colcon-common-extensions
+```
+
+## Getting the sources
+
+The first step is to create a developer workspace in which `vcstool` and
+`colcon` can work:
+
+```bash
+mkdir -p ~/workspace/src
+cd ~/workspace/src
+```
+
+All the sources of gazebo-harmonic are declared in a yaml file. Download
+it to the workspace:
+
+```bash
+wget https://raw.githubusercontent.com/gazebo-tooling/gazebodistro/master/collection-harmonic.yaml
+```
+
+Use `vcstool` to automatically retrieve all the Gazebo libraries sources from
+their repositories:
+
+```bash
+vcs import < collection-harmonic.yaml
+```
+
+The src subdirectory should contain all the sources ready to be built.
+
+
+## Install dependencies
+
+Add `osrf/simulation` to Homebrew formulae
+
+```bash
+brew update
+brew tap osrf/simulation
+```
+
+Install all dependencies:
+
+Dependency for Ogre:
+
+```bash
+brew cask install xquartz
+```
+
+General dependencies:
+
+```bash
+brew install assimp boost bullet cmake cppzmq dartsim@6.10.0 doxygen eigen fcl ffmpeg flann freeimage freetype gflags google-benchmark gts ipopt irrlicht jsoncpp libccd libyaml libzzip libzip nlopt ode open-scene-graph ossp-uuid ogre1.9 ogre2.2 pkg-config protobuf qt qwt rapidjson ruby tbb tinyxml tinyxml2 urdfdom zeromq
+```
+
+`dartsim@6.10.0` and `qt5` are not sym-linked. To use those dependencies when building
+`gazebo-physics2` and `gazebo-gui3`, run the following after installation to add them to `/use/local`:
```

Comment:
remove this?

Reply:
actually I had to do this on my mac. I expanded the instructions to include paths for arm-based Apple silicon processors. [5f8e3e2](https://github.com/gazebosim/docs/pull/324/commits/5f8e3e20eb115262e864f8d151a85f3cf7852096)